percentage-fee-handler/src/lib.rs

```diff
@@ -0,0 +1,263 @@
+// The Licensed Work is (c) 2022 Sygma
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use self::pallet::*;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+pub mod weights;
+pub use weights::*;
+
+#[cfg(test)]
+mod mock;
+
+#[allow(unused_variables)]
+#[allow(clippy::large_enum_variant)]
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::{dispatch::DispatchResult, pallet_prelude::*, traits::StorageVersion};
+	use frame_system::pallet_prelude::*;
+	use sp_std::boxed::Box;
+	use sygma_traits::{DomainID, FeeHandler};
+	use xcm::latest::{AssetId, Fungibility::Fungible, MultiAsset};
+
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+	/// Mapping fungible asset id with domain id to fee rate
+	#[pallet::storage]
+	pub type AssetFeeRate<T: Config> = StorageMap<_, Twox64Concat, (DomainID, AssetId), u32>;
```

Comment:
I have been thinking about this suggestion, and here is my thoughts:
1. there is a get_fee trait which take a domainID and a multiAsset, this trait is impl in basic fee handler pallet, percentage fee handler pallet and fee handler router pallet.
2. pallets do not know anything about the resource, they only know multiasset and domain, because in the bridge pallet we register domains. Resource is used to bind asset multilocation in each runtime. with this, if we want to bind resource with fee rate cross all domains, we need to move resource to pallet level, which takes a lot refactoring.
3. domain only exists in the bridge pallet storage currently, which means all fee related pallets do not have access to this storage, domain+assetID is just a key to store fee amount or fee rate in fee related pallet, with this, when setting fee, it is hard to set it for all domains, unless we build the logic to read bridge pallet storage from other pallet, which again might take a lot effort
4. Current fee design in substrate is requiring some admin extrnisics to set fee, which is a bit repeating for sure, it works well so far because we have only one domain from substrate point of view, I guess we need to plan and design a bit and impl such big changes in another PR if we decide to.

Reply:
I'm going to mark this one as resolved and merge this PR, but thanks for the suggestions. Let's continue this conversation in sprint plan and maybe impl later